test/test_autograd.py

```diff
@@ -2094,6 +2094,16 @@ def test_cat_empty_legacy(self):
                               False, f_args_variable, f_args_tensor)
         self.assertTrue(gradcheck(lambda a, b: torch.cat((a, b)), f_args_variable, eps=1e-6, atol=PRECISION))
 
+    def test_cdist(self):
+        for p in [0, 1, 2, 3, 1.5, 2.5, float('inf')]:
+            f_args_variable = (torch.randn(S, S, requires_grad=True),
+                               torch.randn(S, S, requires_grad=True))
+            f = lambda a, b: torch.cdist(a, b, p)
+            f_args_tensor = deepcopy(unpack_variables(f_args_variable))
+            run_functional_checks(self, "test_cdist", "cdist", f,
+                                  True, f_args_variable, f_args_tensor)
```

Comment:
hmm...this exposed a pretty serious bug in our testing.  This test is running double backwards tests and passing, but it shouldn't be because you didn't write a double backwards function.

I believe what is happening is this:
1) because you don't have an entry in derivatives.yaml for your backwards function (i.e. a double backwards definition), it is assumed to be composed of differentiable functions
2) it's not actually composed of differentiable functions
3) the testing doesn't catch this because it only looks at differentiable outputs.  But your function doesn't create any differentiable outputs.

This should be testable, though.  Basically, this should only be valid if the numerical gradient of the function is 0.  Can you add checks for that?  (You should probably do it in a separate commit, there may be other functions that exhibit the same problem).

Reply:
Added test to EXCLUDE_GRADGRADCHECK_BY_TEST_NAME list